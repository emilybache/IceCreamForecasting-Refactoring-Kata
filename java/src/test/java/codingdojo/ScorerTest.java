package codingdojo;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Objects;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ScorerTest {
    @Test
    void getScore() {
        Boolean[] sunnyTodays = new Boolean[]{true, false};
        String[] flavours =  Arrays.stream(IceCream.values())
                .map(Enum::name)
                .toList()
                .toArray(new String[0]);

        CombinationApprovals.verifyAllCombinations(
                ScorerTest::doGetScore,
                sunnyTodays,
                flavours
                );
    }


    static String doGetScore(Boolean sunnyToday, String flavourName) {
        Scorer.flavour = IceCream.valueOf(flavourName);
        int actual = Scorer.getScoreWithWeather(sunnyToday);
        return "" + actual;
    }

}
