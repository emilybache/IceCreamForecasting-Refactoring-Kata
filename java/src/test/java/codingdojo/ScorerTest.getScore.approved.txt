[true, Strawberry] => 10 
[true, Chocolate] => 6 
[true, Vanilla] => 7 
[false, Strawberry] => 5 
[false, Chocolate] => 6 
[false, Vanilla] => 5 
